### DISCUSSION TITLE
How is storage in the warehouse managed?

### DISCUSSION THREAD

**Discussion by FabianoVeglianti** [Keriew/augustus#874] [Q&A]
I was reading the code to understand how it works and maybe in the future contribute to Augustus, but I'm having trouble understanding how storage in warehouse is managed: specifically, I'm having trouble understanding the `building_warehouse_add_resource` function in `augustus/src/building/warehouse.c`. Why you can put the resource in the warehouse ` if (space->loads_stored < 4) `, Where do you come up with that 4?

**Comment by crudelios** [comment]
Hey there! Glad to see new people interested in the code.

Each warehouse building is actually composed of 9 1x1 sub-buildings: 1 "crane" tile and 8 storage tiles.

As you can see visually, each space tile can store at most 4 units.

The code you're looking at makes sure no more than 4 units get stored per each warehouse space tile.
